src/index.js

```diff
@@ -54,226 +56,301 @@ function buildNametoIdMap(objectArr, type) {
-      if (!transformationDict.hasOwnProperty(transformerVersionID)) {
-        continue;
-      }
+    let expectedOutputfile =
+      transformationDict[transformerVersionID]["expected-output"];
+    let expectedOutput = expectedOutputfile
+      ? JSON.parse(fs.readFileSync(expectedOutputfile))
+      : "";
 
-      const apiOutput = successResults[i].result.output.transformedEvents;
-      const transformationName = transformationDict[transformerVersionID].name;
-      const transformationHandleName = _.camelCase(transformationName);
+    if (expectedOutput == "") {
+      continue;
+    }
 
-      fs.writeFileSync(
-        `${testOutputDir}/${transformationHandleName}_output.json`,
-        JSON.stringify(apiOutput, null, 2)
+    if (!isEqual(expectedOutput, apiOutput)) {
+      core.info(
+        `Test output do not match for transformation: ${transformationName}`
       );
-      testOutputFiles.push(
-        `${testOutputDir}/${transformationHandleName}_output.json`
+      outputMismatchResults.push(
+        `Test output do not match for transformation: ${transformationName}`
       );
 
-      if (
-        !transformationDict[transformerVersionID].hasOwnProperty(
-          "expected-output"
-        )
-      ) {
-        continue;
-      }
-
-      let expectedOutputfile =
-        transformationDict[transformerVersionID]["expected-output"];
-      let expectedOutput = expectedOutputfile
-        ? JSON.parse(fs.readFileSync(expectedOutputfile))
-        : "";
-
-      if (expectedOutput == "") {
-        continue;
-      }
-
-      if (!isEqual(expectedOutput, apiOutput)) {
-        core.info(
-          `Test output do not match for transformation: ${transformationName}`
-        );
-        outputMismatchResults.push(
-          `Test output do not match for transformation: ${transformationName}`
-        );
-
-        fs.writeFileSync(
-          `${testOutputDir}/${transformationHandleName}_diff.json`,
-          JSON.stringify(detailedDiff(expectedOutput, apiOutput), null, 2)
-        );
-
-        testOutputFiles.push(
-          `${testOutputDir}/${transformationHandleName}_diff.json`
-        );
-      }
-    }
+      fs.writeFileSync(
+        `${testOutputDir}/${transformationHandleName}_diff.json`,
+        JSON.stringify(detailedDiff(expectedOutput, apiOutput), null, 2)
+      );
 
-    // upload artifact
-    if (uploadTestArtifact === "true") {
-      core.info("Uploading test api output...");
-      await artifactClient.uploadArtifact(
-        "transformer-test-results",
-        testOutputFiles,
-        "."
+      testOutputFiles.push(
+        `${testOutputDir}/${transformationHandleName}_diff.json`
       );
     }
+  }
+}
+
+// Upload the test results to an artifact store.
+async function uploadTestArtifacts(testOutputFiles) {
+  // upload artifact
+  if (uploadTestArtifact) {
+    core.info("Uploading test api output");
+    await artifactClient.uploadArtifact(
```

Comment:
Please make sure you are checking the response of artificatClient.uploadArtifact and in case the failedItems exist raise an Exception